### DISCUSSION TITLE
Building of Faunus on Linux cluster/desktop

### DISCUSSION THREAD

**Discussion by directysj** [mlund/faunus#389] [Q&A]
Hello,

Can I ask if anybody build `Faunus 2.4.2` version onward on a Linux cluster/desktop? 

Thank you in advance!!

Best regards,
Jalim Singh

**Comment by directysj** [comment]
> Hello,
> 
> Can I ask if anybody build `Faunus 2.4.2` version onward on a Linux cluster/desktop?
> 
> Thank you in advance!!
> 
> Best regards, Jalim Singh
Hello @mlund 

Sorry for late reply. Finally, I could install Faunus 2.5.0 using GCC10.3.0 on my Linux machine where I don't have root access. GCC10+ module, which was having problem in gathering some parallel libraries support. This was solved by editing the CMakeLists.txt file. Further, I was getting other errors, which I searched on Google and found that GCC 9 and 10 does not have some library support for TBB, then I disabled it in CMakeLists.txt and tell it to not to search with GCC instead in TBB ../lib directory. This leads to successful building of the Faunus.  Two edits in the CMakeLists.txt are shown below:

```
add_definitions(-D_GLIBCXX_USE_C99_MATH=1)
add_definitions(-D_GLIBCXX_USE_TBB_PAR_BACKEND=0)
```

In addition to this, let me point out that installation of the Faunus directly from Conda only gives its 2.4.0 version, which does not have OpenMP/TBB parallelization. I don't know why it does not add the latest version.

Thank you
~ Jalim

**Reply by mlund** [reply]
Thanks for posting your solution here; I completely missed this post and only saw it now. This was/is a known problem with some GCC versions that oddly rely on TBB (but do not bundle it). Another trick is to add `-DENABLE_OPENMP=false` to CMake.
I think the compilation should be smoother on newer compiler and Faunus combinations.
